src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerWaitContainer.groovy

```diff
@@ -15,19 +15,37 @@
  */
 package com.bmuschko.gradle.docker.tasks.container
 
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+
+import java.util.concurrent.TimeUnit
+
 class DockerWaitContainer extends DockerExistingContainer {
 
     protected int exitCode
 
-    DockerWaitContainer(){
-        ext.getExitCode = {exitCode}
+    /**
+     * Wait timeout in seconds.
+     */
+    @Input
+    @Optional
+    Long timeout
+
+    DockerWaitContainer() {
+        ext.getExitCode = { exitCode }
     }
 
     @Override
     void runRemoteCommand(dockerClient) {
-        logger.quiet "Waiting for container with ID '${getContainerId()}'."
+        logger.quiet "Waiting for container with ID '${getContainerId()}' for ${timeout}s."
```

Comment:
Let's do something like the below. For most cases this may be null and seeing the `null` within the output may give a false impression that a user has done something wrong.


    String possibleTimeout = timeout ? " for ${timeout} milliseconds" : ''
    logger.quiet "Waiting for container with ID '${getContainerId()}'${possibleTimeout}."